connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorHeartbeatConnector.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.mirror;
+
+import org.apache.kafka.connect.connector.Task;
+import org.apache.kafka.connect.source.SourceConnector;
+import org.apache.kafka.common.config.ConfigDef;
+
+import java.util.Map;
+import java.util.List;
+import java.util.Collections;
+
+
+public class MirrorHeartbeatConnector extends SourceConnector {
+    private MirrorConnectorConfig config;
+
+    @Override
+    public void start(Map<String, String> props) {
+        config = new MirrorConnectorConfig(props);
```

Comment:
@vpernin This behavior is subtle but correct. We want heartbeats going everywhere, even to clusters that aren't a target of any source->target replication. This is because we need heartbeats to exist upstream in order to replicate them downstream. If we are replicating A->B, we don't want to emit heartbeats only to B -- that wouldn't really tell us much, except that MM can send to B. What we want to know is how long it takes records to travel from A to B. So we emit heartbeats to A and _replicate them_ to B. This lets us monitor latency between A and B even when no other records are being replicated.

Specifically, we create herders between every pair of clusters (a fully-connected mesh) and emit heartbeats everywhere. Then, for the subset of "enabled" replications, we start at least one MirrorSourceConnector task. These tasks _always_ replicate heartbeats, so the result is "A.heartbeat" in cluster B whenever A->B is enabled.

We could _only_ emit records upstream (i.e. only to sources and not targets) and achieve this same result, but heartbeats are useful for more than measuring cross-cluster latency. Emitting heartbeats everywhere makes various other tooling possible. For example, you can query any single cluster and find out about every other cluster just by consuming the heartbeat topic, since heartbeats will have come from everywhere.

Reply:
Ok, I understand this conception.
Maybe, this clear explanation has its place in the KIP and the documentation.